File: txt_photo.c
#include <stdio.h>
#include <stdlib.h>

int main(int argc, char* argv[])
{
	if (argc < 2)
	{
		printf("please enter parameter!\n");
		return 3;
	}
	int ch;
	FILE* src;
	FILE* export;
	const char convert_ch[14] = {' ', '.', '`', ':', '~', '*', '=', '&', '%', '#'};
	src = fopen(argv[1], "rb");
	export = fopen("export.txt", "w");
	if (export == NULL)
	{
		printf("export is failed!\n");
		exit(4);
	}
	if (src == NULL)
	{
		printf("open source-file is failed!\n");
		exit(1);
	}
	ch = getc(src);
	while (ch != EOF)
	{
		printf("%c", convert_ch[ch-48]);                      
		putc(convert_ch[ch-48], export);
		ch = getc(src);
		while (ch == 32 || ch == 13 || ch == 10)    /*如获取到空格则获取下一个*/
		{
			if (ch == 10)
			{
				putc(10, export);
			}
			ch = getc(src);
		}
	}
	if (fclose(src) != 0)                /*文件关闭及相关提示*/
	{
		printf("\nError in closing source-file \n");
		exit(2);
	}
	if (fclose(export) != 0)
	{
		printf("\nError in closing output-file \n");
		exit(2);
	}
	printf("\nclose file comlete!\n");
	exit(0);
}

File: txt_photo.py
import sys


def main():
    namespace = locals()
    with open(sys.argv[1], "r") as f:
        with open("output.txt", "w") as save:
            content = f.read()
            content = content.replace(" ", "")
            convert = [" ", ".", "`", ":", "~", "*", "=", "&", "%", "#"]
            x = 0
            namespace["content_0"] = []
            for ch in content:
                if ch == "\n":
                    x = x + 1
                    namespace["content_" + str(x)] = []
                    save.write("\n")
                    continue
                try:
                    ch = convert[int(ch)]
                except IndexError:
                    raise ValueError
                save.write(ch)
                print(ch, end="")
                namespace["content_" + str(x)].append(ch)
    return 0


if __name__ == "__main__":
    try:
        main()
    except IndexError:
        print("please enter parameter!\n")
        exit(2)
    except FileNotFoundError:
        print("file not found!\n")
        exit(3)
    except ValueError:
        print("the numbers in file is unsupported!\n")
        exit(4)
